monitor/flowgen/flowgen.go

```diff
@@ -64,7 +90,8 @@ func Start(
 					return
 				case <-timer.C:
 					jobsTotal.Inc()
-					if err := run(log.WithCtx(ctx, "job", job.Name), backends, job); err != nil {
+					ctx := log.WithCtx(ctx, "job", job.Name)
```

Comment:
assigning ctx inside the loop is risky, are you sure you are not overriding the ctx for other loop iterations?